### PR TITLE
KSW/add annotation shortcut

### PR DESCRIPTION
**Pull request by kswang1029**
**Description**
This is to address #2167 

A new AnnotationMenuComponent is created and applied in the ToolbarComponent.
Then a new annotation shortcut button is added to the ToolbarMenuComponent and reuse the AnnotationMenuComponent to display all annotation types for selection and draw.
If the new annotation shortcut button is clicked, the annotation type menu will show up. If no annotation is selected, the default is the point annotation.

**Checklist**

For linked issues (if there are):
- [x] assignee and label added
- [x] ZenHub issue connection, board status, and estimate updated

For the pull request:
- [x] reviewers and assignee added
- [x] ZenHub estimate, milestone, and release (if needed) added
- [x] e2e test passing / ~corresponding fix added~
- [x] changelog updated / ~no changelog update needed~
- [x] ~protobuf updated to the latest dev commit~ / no protobuf update needed
- [x] `BackendService` unchanged / ~`BackendService` changed and corresponding ICD test fix added~

### CODE REVIEW

File: src/components/Shared/AnnotationMenu/AnnotationMenuComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import {CARTA} from "carta-protobuf";
+import {observer} from "mobx-react";
+
+import {CustomIcon, CustomIconName} from "icons/CustomIcons";
+import {AppStore} from "stores";
+import {RegionMode, RegionStore} from "stores/Frame";
+
+@observer
+export class AnnotationMenuComponent extends React.Component {
+    handleRegionTypeClicked = (type: CARTA.RegionType) => {
```
Comment (crocka) [inline src/components/Shared/AnnotationMenu/AnnotationMenuComponent.tsx]: Maybe we can pass this function in as a prop?

File: src/components/Menu/ToolbarMenu/ToolbarMenuComponent.tsx [hunk 2]
```diff
@@ -105,6 +106,13 @@ export class ToolbarMenuComponent extends React.Component {
                 </i>
             </span>
         );
+
+        const annotationMenu = (
+            <Menu style={{padding: 0}}>
```
Comment (YuHsuan-Hwang) [inline src/components/Menu/ToolbarMenu/ToolbarMenuComponent.tsx]: I think the dropdown looks better without removing the padding.

File: src/components/Menu/ToolbarMenu/ToolbarMenuComponent.tsx [hunk 3]
```diff
@@ -117,6 +125,27 @@ export class ToolbarMenuComponent extends React.Component {
                             </Tooltip2>
                         );
                     })}
+
+                    <Popover2 popoverClassName="annotation-menu" content={annotationMenu} position={Position.BOTTOM} minimal={true} disabled={regionButtonsDisabled}>
```
Comment (YuHsuan-Hwang) [inline src/components/Menu/ToolbarMenu/ToolbarMenuComponent.tsx]: `popoverClassName` can be removed if it's not used. I would prefer we set `Position.BOTTOM_LEFT` to be consistent with the menu.

File: src/components/Menu/ToolbarMenu/ToolbarMenuComponent.tsx [hunk 4]
```diff
@@ -117,6 +125,27 @@ export class ToolbarMenuComponent extends React.Component {
...
+                                    Annotation <br /> <small>Click to select annotation type.</small>
+                                </span>
+                            }
+                            position={Position.BOTTOM}
+                        >
+                            <AnchorButton
+                                icon={"annotation"}
+                                disabled={regionButtonsDisabled}
+                                active={isRegionCreating === true && appStore.activeFrame.regionSet.isNewRegionAnnotation === true}
+                                onClick={() => {
```
Comment (YuHsuan-Hwang) [inline src/components/Menu/ToolbarMenu/ToolbarMenuComponent.tsx]: Maybe we don't need to enable annotation creation mode by clicking the button. The user could be just checking what types of annotations are available.

File: src/components/Shared/AnnotationMenu/AnnotationMenuComponent.tsx (hunk 1 above)
Comment (YuHsuan-Hwang) [inline src/components/Shared/AnnotationMenu/AnnotationMenuComponent.tsx]: Perhaps we can set `handleRegionTypeClicked` as a prop and reuse `handleRegionTypeClicked` in `ToolbarComponent` and `ToolbarMenuComponent`. These two functions are actually slightly different. I think we can try to use functional components to make the code simpler.

**Review by YuHsuan-Hwang — CHANGES_REQUESTED** [review]
Tested that the shortcut works good 👍  I have some suggestions:

File: src/components/Menu/ToolbarMenu/ToolbarMenuComponent.tsx (hunk 4 above)
Comment (kswang1029) [reply src/components/Menu/ToolbarMenu/ToolbarMenuComponent.tsx]: the idea behind this is to sync the active state of this shortcut button with the region button in the toolbar. also to be consistent with the other region shortcut buttons about the active state. will think about how to simplify this.

**Comment by kswang1029** [comment]
@YuHsuan-Hwang I have tried to refactor/fix the parts that you commented. Could you have another look to make sure the direction is right?

**Comment by YuHsuan-Hwang** [comment]
@kswang1029 The direction is right 👍

File: src/components/ImageView/Toolbar/ToolbarComponent.tsx [hunk 5]
```diff
@@ -147,11 +147,7 @@ export class ToolbarComponent extends React.Component<ToolbarComponentProps> {
 
         const annotationMenu = (
             <Menu style={{padding: 0}}>
-                {Array.from(RegionStore.AVAILABLE_ANNOTATION_TYPES).map(([type, text], index) => {
-                    const annotationIconString: IconName | CustomIconName = RegionStore.RegionIconString(type);
-                    const annotationIcon = RegionStore.IsRegionCustomIcon(type) ? <CustomIcon icon={annotationIconString as CustomIconName} /> : (annotationIconString as IconName);
-                    return <MenuItem icon={annotationIcon} text={text} onClick={() => this.handleRegionTypeClicked(type)} key={index} />;
-                })}
+                <AnnotationMenuComponent handleRegionTypeClicked={type => this.handleRegionTypeClicked(type)} />
```
Comment (YuHsuan-Hwang) [inline src/components/ImageView/Toolbar/ToolbarComponent.tsx]: Can be simplified to `{this.handleRegionTypeClicked}`.

File: src/components/Menu/ToolbarMenu/ToolbarMenuComponent.tsx [hunk 6]
```diff
@@ -105,6 +106,13 @@ export class ToolbarMenuComponent extends React.Component {
                 </i>
             </span>
         );
+
+        const annotationMenu = (
+            <Menu>
+                <AnnotationMenuComponent handleRegionTypeClicked={type => this.handleRegionTypeClicked(type)} />
```
Comment (YuHsuan-Hwang) [inline src/components/Menu/ToolbarMenu/ToolbarMenuComponent.tsx]: Can be simplified to `{this.handleRegionTypeClicked}`.

File: src/components/Shared/AnnotationMenu/AnnotationMenuComponent.tsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+import {CARTA} from "carta-protobuf";
+
+import {CustomIcon, CustomIconName} from "icons/CustomIcons";
+import {RegionStore} from "stores/Frame";
+
+interface AnnotationMenuComponentProps {
+    handleRegionTypeClicked: (type: CARTA.RegionType) => void;
+}
+
+export const AnnotationMenuComponent: React.FC<AnnotationMenuComponentProps> = ({handleRegionTypeClicked}) => {
```
Comment (YuHsuan-Hwang) [inline src/components/Shared/AnnotationMenu/AnnotationMenuComponent.tsx]: `React.FC` could be problematic when upgrading to React 18. Change it to:
`export const AnnotationMenuComponent = ({handleRegionTypeClicked}: AnnotationMenuComponentProps) => {`.

File: src/components/Shared/AnnotationMenu/AnnotationMenuComponent.tsx [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    return (
+        <React.Fragment>
```
Comment (YuHsuan-Hwang) [inline src/components/Shared/AnnotationMenu/AnnotationMenuComponent.tsx]: We could simplify it to `<> ... </>`.

**Review by YuHsuan-Hwang — CHANGES_REQUESTED** [review]
The new changes are good. I have some final comments.

**Comment by kswang1029** [comment]
@YuHsuan-Hwang comments are addressed. thanks.
 No regression from e2e test apparently.

**Review by YuHsuan-Hwang — APPROVED** [review]

**Review by crocka — APPROVED** [review]

**Review by acdo2002 — APPROVED** [review]
Checking the ICD working flow, the shortcut performance is what I have expected.

**Pull request merged**